### ISSUE TITLE
slickr tips

### ISSUE THREAD

**Issue by yonicd** [dan-reznik/Elliptical-Billiards-Triangular-Orbits#1]
very nice use of slickR!

for the heavier carousels you can use the lazy loading option. that will alleviate the pressure off the html.

last comment in the details block.

https://metrumresearchgroup.github.io/slickR/reference/slickR.html#details

**Comment by yonicd** [comment]
I also think you meant in the envelope child to use these settings. The autoplay has to be TRUE for the speed to be used.

`slickR::settings(autoplay = TRUE, autoplaySpeed = 100, fade = TRUE, adaptiveHeight = TRUE)`

**Comment by dan-reznik** [comment]
Hi Yoni

I set am now using

slickR(
    <<... large set of png's...>,
    height=553,
    width = 800, slideType = 'img-lazy') +
  slickR::settings(fade=T,adaptiveHeight=T,lazyLoad = 'ondemand')

However the size of my html is still enormous, is bringing in all pictures.
Should I use perhaps URLs instead of filenames above?

Dan

**Comment by yonicd** [comment]
yes. that would be much lighter, and i think that lazyload works with urls and not embedded.

**Comment by yonicd** [comment]
also try letting height be set internally and set width to "95%". that may get the sizing to react better

**Comment by dan-reznik** [comment]
but the question is -- what can i do so rmarkdown doesn't embed the images
(encoded in ASCII) into the output html? Right now the .html contains all
images and its size is 50 Mb!

Dan

**Comment by yonicd** [comment]
rmarkdown embeds on default. i went down that rabbit hole a while back. if you pass in url's it still forces an embed?

i use urls in the basics vignette of slickr and the size is not inflated. i have a ton of image calls in that file.

**Comment by yonicd** [comment]
If the images (not in slickr) are local you could set embed to false and the html should link to the images i think. Here is a link in SO explaining how to do it. https://stackoverflow.com/questions/14870589/knitr-r-how-not-to-embed-images-in-the-html-file

**Comment by dan-reznik** [comment]
no, passing in the URL solved the problem! the tricky thing is for the svgs
to be out on the server already else you can't knit it

thanks so much!

one last thing: in my page several slickR carousels are behind "tabsets" --
any idea on how to make them not look empty when I first select the tabset?
I.e., some JS trick to preinitialize them?

Dan

**Comment by yonicd** [comment]
hmmm not sure. maybe @colinfay may know a trick for that

**Comment by yonicd** [comment]
i would though still replace `speed` with `autoplay = TRUE, autoplaySpeed = 100`. I dont think `speed` actually does anything, at least in the html you have with the carousels they are not moving. if it is autoplaying that may get the slick to activate when you move tabs

**Comment by dan-reznik** [comment]
when you are not in "autoplay" speed regulates how fast you can jump from
one slide to the other

Dan

**Comment by yonicd** [comment]
you read the manual better than me :)
